compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -95,21 +95,50 @@ static llvm::cl::opt<bool> clNormalizeInputIndexingMap(
     llvm::cl::desc("Enable normalizing input indexing map to identity"),
     llvm::cl::init(false));
 
-static llvm::cl::opt<bool> clDumpDispatchGraph(
-    "iree-flow-dump-dispatch-graph",
-    llvm::cl::desc("Dump a dot graph for dispatches"), llvm::cl::init(false));
-
-static llvm::cl::opt<std::string> clDumpDispatchGraphOutputFile(
-    "iree-flow-dump-dispatch-graph-output-file",
-    llvm::cl::desc("Output file name for a dispatch graph dump"),
-    llvm::cl::init("dispatch.dot"));
-
 static llvm::cl::opt<std::string> clDispatchTransformFileName(
     "iree-flow-dispatch-use-transform-dialect",
     llvm::cl::desc("mlir file containing a top-level module that specifies "
                    "the transformations to apply to form dispatch regions."),
     llvm::cl::init(""));
 
+static llvm::cl::opt<bool> clDumpGraphAfterOutlining(
+    "iree-flow-dump-graph-after-outlining",
+    llvm::cl::desc("Dump a graph after outlining"), llvm::cl::init(false));
+
+static llvm::cl::opt<std::string> clDumpGraphAfterOutliningOutputFile(
+    "iree-flow-dump-graph-after-outlining-output-file",
+    llvm::cl::desc("Output file name for the graph after outlining"),
+    llvm::cl::init("graph-after-outlining.dot"));
+
+static llvm::cl::opt<bool> clDumpGraphAfterDeduplication(
+    "iree-flow-dump-graph-after-deduplication",
+    llvm::cl::desc("Dump a graph after deduplication"), llvm::cl::init(false));
+
+static llvm::cl::opt<std::string> clDumpGraphAfterDeduplicationOutputFile(
+    "iree-flow-dump-graph-after-deduplication-output-file",
+    llvm::cl::desc("Output file name for the graph after deduplication"),
+    llvm::cl::init("graph-after-deduplication.dot"));
+
+static llvm::cl::opt<bool> clDumpStatsAfterOutlining(
+    "iree-flow-dump-stats-after-outlining",
+    llvm::cl::desc("Dump operation statistis after outlining dispatches"),
+    llvm::cl::init(false));
+
+static llvm::cl::opt<std::string> clDumpStatsAfterOutliningOutputtFile(
+    "iree-flow-dump-stats-after-outlining-output-file",
+    llvm::cl::desc("Output file name for the statistics after outlining"),
+    llvm::cl::init("stats-after-outlining.json"));
+
+static llvm::cl::opt<bool> clDumpStatsAfterDeduplication(
+    "iree-flow-dump-stats-after-deduplication",
+    llvm::cl::desc("Dump operation statistis after deduplication dispatches"),
```

Comment:
```suggestion
    llvm::cl::desc("Dump operation statistics after deduplication dispatches"),
```
(vscode and other editors can have spell checkers enabled, which makes these things easier to avoid)